batching-gson/src/main/java/com/flipkart/batching/gson/GsonSerializationStrategy.java

```diff
@@ -252,8 +249,8 @@ public void build() {
         GsonBuilder gsonBuilder = new GsonBuilder();
         gsonBuilder.registerTypeAdapterFactory(dataAdapter);
         gsonBuilder.registerTypeAdapterFactory(batchInfoAdapter);
```

Comment:
This was used to register the sub classes of Batch, and Data class with gson. We can remove this since all the internal classes have been moved to type adapters, and has been registered with the factory. 

But for the subtypes that the client might create for these classes, will it be a right idea to mandate to take in the type adapter for the subclass as well